### ISSUE TITLE
DateFormat.parseLoose() requires trailing whitespace on delimiters

### ISSUE THREAD

**Issue by cpelling** [dart-lang/intl#81]
``` dart
// assuming en-US locale
new DateFormat.yMMMd().parseLoose('Jan 23, 2015'); // OK
new DateFormat.yMMMd().parseLoose('Jan 23,2015'); // throws FormatException
```

I would expect the second call to give the same result as the first. Admittedly, DateFormat.parseLoose()'s documentation is strictly speaking ambiguous on this point (**bold** is mine):

> If [inputString] is accepted by [parseStrict], just return the result. If not, attempt to parse it, but accepting either upper or lower case, allowing delimiters to be missing and replaced or supplemented with whitespace, and **_allowing arbitrary amounts of whitespace wherever whitespace is permitted**_.

I would argue that "arbitrary amounts of whitespace" should include "no whitespace".

**Comment by cpelling** [comment]
The following changes to [_DateFormatField.parseLiteralLoose()](https://github.com/dart-lang/intl/blob/1c2757633fb7a425a4ca48eb7902dbdf4960aa2c/lib/src/intl/date_format_field.dart#L61) appear to fix issue #80 and this issue.

```dart
  void parseLiteralLoose(_Stream input) {
    var found = input.peek(width);
    var trimmedPattern = pattern.trim();
    if (found.startsWith(trimmedPattern)) {
      input.read(trimmedPattern.length);
      while (!input.atEnd() && input.peek().trim().isEmpty) {
        input.read();
      }
    } else {
      throwFormatException(input);
    }
  }
```

**Comment by alan-knight** [comment]
Fixed in https://github.com/dart-lang/intl/commit/0c284428b4d2b97e944e29e9b5f2c5a5deea39af
